std/algorithm/searching.d

```diff
@@ -3462,6 +3462,63 @@ unittest
 }
 
 /**
+Computes the index of the first occurrence of `range`'s minimum (respectively maximum).
+
+Params:
+    pred = The ordering predicate to use to determine the extremum (minimum or
+           maximum) element.
+    range = The input range to search.
+
+Returns:
+    The index of the first encounter of the minimum (respectively maximum) in `range`. If the
+    range is empty, -1 is returned.
+ */
+Signed!size_t minPosIndex(alias pred = "a < b", Range)(Range range)
+    if (isForwardRange!Range && !isInfinite!Range &&
+        is(typeof(binaryFun!pred(range.front, range.front)) == bool))
+{
+    if (range.empty) return -1;
+
+    Signed!size_t minPos = 0, curPos = 0;
+    auto minFront = range.save;
```

Comment:
The common use case is `auto amax = range.map!abs.maxIndex;`. OK lets use `.front`